aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -361,6 +361,9 @@ module aptos_framework::coin {
     /// "Merges" the two given coins.  The coin passed in as `dst_coin` will have a value equal
     /// to the sum of the two tokens (`dst_coin` and `source_coin`).
     public fun merge<CoinType>(dst_coin: &mut Coin<CoinType>, source_coin: Coin<CoinType>) {
+        spec {
+            assume dst_coin.value + source_coin.value <= MAX_U64;
```

Comment:
I would mark a TODO item here, with probably a better way of specifying this as a struct invariant, e.g.,
```
spec Coin {
    invariant value <= SOME_PREDEFINED_MAX_VALUE_BUT_WAY_SMALLER_THAN_MAX_U64;
}
```

Reply:
This is actually a good point. We assume that merging two `Coin<AptosCoin>` will never abort based on our knowledge on the `AptosCoin` operation/governance. However, it may not be true in general. For example, a user defined coin can be minted and merged in an arbitrary way which can causes an arithmetic overflow, aborting with an internal error.

Reply:
A struct invariant sounds nice! To be able to prove that, we may also need to put some assertions (with a proper error code) in the code.